matminer/featurizers/base.py

```diff
@@ -364,10 +364,28 @@ def featurize_many(self, entries, ignore_errors=False, return_errors=False,
                                            return_errors=return_errors,
                                            pbar=pbar)
             with Pool(self.n_jobs) as p:
-                func = partial(self.featurize_wrapper,
+                func = partial(self.featurize_imap_wrapper,
                                return_errors=return_errors,
                                ignore_errors=ignore_errors)
-                return p.map(func, entries, chunksize=self.chunksize)
+                entries = list(
+                    enumerate(entries))  # turns entries into [(0, entry_1), (1, entry_2), (2, entry_3), ... ]
+                results = []
+                for i, result in tqdm_func(
+                        p.imap_unordered(func, entries, chunksize=self.chunksize), total=len(entries)):
+                    results[i] = result
+                return results
+
+    def featurize_imap_wrapper(self, data):
+        """
+        A helper function to featurize the tuple form data
```

Comment:
Can you make this doc string explain what it is doing. E.g. it accepts the index of the entry and the entry and returns the index and the features.

Reply:
Is there a reason we can't just replace featurize_wrapper with featurize_imap_wrapper? It seems like a bit much to have one featurize wrapper for non-tqdm featurization and a different featurize_wrapper for tqdm featurization...

Reply:
I think a wrapper of this sort will be the cleanest way to achieve this. However, I think your lambda idea will tidy it up further.